include/util/io.hpp

```diff
@@ -0,0 +1,70 @@
+#ifndef OSRM_INCLUDE_UTIL_IO_HPP_
+#define OSRM_INCLUDE_UTIL_IO_HPP_
+
+#include "util/simple_logger.hpp"
+
+#include <boost/filesystem.hpp>
+#include <boost/filesystem/fstream.hpp>
+
+#include <cstddef>
+#include <cstdint>
+
+#include <bitset>
+#include <vector>
+
+namespace osrm
+{
+namespace util
+{
+
+inline bool serializeFlags(const boost::filesystem::path &path, const std::vector<bool> &flags)
+{
+    // TODO this should be replaced with a FILE-based write using error checking
+    std::uint32_t number_of_bits = flags.size();
+    boost::filesystem::ofstream flag_stream(path, std::ios::binary);
+    flag_stream.write(reinterpret_cast<const char *>(&number_of_bits), sizeof(number_of_bits));
+    // putting bits in ints
+    std::uint32_t chunk = 0;
+    std::size_t chunk_count = 0;
+    for (std::size_t bit_nr = 0; bit_nr < number_of_bits;)
+    {
+        std::bitset<32> chunk_bitset;
+        for (std::size_t chunk_bit = 0; chunk_bit < 32 && bit_nr < number_of_bits;
+             ++chunk_bit, ++bit_nr)
+            chunk_bitset[chunk_bit] = flags[bit_nr];
+
+        chunk = chunk_bitset.to_ulong();
+        ++chunk_count;
+        flag_stream.write(reinterpret_cast<const char *>(&chunk), sizeof(chunk));
+    }
+    util::SimpleLogger().Write() << "Wrote " << number_of_bits << " bits in " << chunk_count
+                                 << " chunks (Flags).";
+    return !flag_stream.fail();
```

Comment:
Hmm I would be careful with explicitely checking states here; if you just want to verify the I/O operation went fine and the stream is okay, then force the stream to a boolean context, such as `if(stream) ok();`